appknox/client.py

```diff
@@ -511,37 +511,31 @@ def rescan(self, file_id: int) -> int:
             raise RescanError('Something went wrong, retry rescan')
         return file
 
-    @lru_cache(maxsize=1)
-    def get_organization_preference(self) -> OrganizationPreference:
+    def get_profile_report_preference(self, profile_id: int) -> ProfileReportPreference:
         """
-        Fetch organization preference for current organization
+        Fetch profile report preference
         """
         try:
-            org_preference = self.drf_api['organizations/{}/preference'.format(self.organization_id)]().get()
+            profile_report_preference = self.drf_api['profiles/{}/report_preference'.format(profile_id)]().get()
         except:
-            raise OrganizationPreferenceError('Could not fetch organization preference')
-        report_pref = org_preference['report_preference']
-        org_report_pref = OrganizationReportPreference(report_pref['show_gdpr'], report_pref['show_hipaa'], report_pref['show_pcidss'])
-        return OrganizationPreference(org_preference['id'], org_report_pref)
-
-    def get_organization_report_preference(self) -> OrganizationReportPreference:
-        """
-        Read report preferences configured at organization level
-        """
-        org_report_pref = self.get_organization_preference().report_preference
-        return org_report_pref
+            raise ProfileReportPreferenceError('Could not fetch profile report preference')
+        show_gdpr = ProfileReportPreferenceConfig(profile_report_preference['show_gdpr']['value'])
+        show_hipaa = ProfileReportPreferenceConfig(profile_report_preference['show_hipaa']['value'])
+        show_pcidss = ProfileReportPreferenceConfig(profile_report_preference['show_pcidss']['value'])
+        return ProfileReportPreference(show_gdpr, show_hipaa, show_pcidss)
```

Comment:
Can you make a class method in the dataclass to generate the ProfileReportPreference instance instead of doing in the client method?

Reply:
Yes, I moved the instantiation to mapper file.